### PR TITLE
add pr-slack-alert action

### PR DESCRIPTION
**Pull request by momentmaker**
This adds a new action called: `pr-slack-alert` which will ping a certain slack user_group for a PR review for a specific slack channel.

Tested:
https://github.com/smartcontractkit/releng-test/actions/runs/9747113303/job/26904262283
https://github.com/smartcontractkit/releng-test/actions/runs/9748784091/job/26904514884

### CODE REVIEW

**Review by chainchad — DISMISSED** [review]

File: actions/pr-slack-alert/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    runs-on: ubuntu-latest
+    env:
+      SLACK_CHANNEL_ID: 7777777777
+      SLACK_USER_GROUP_ID: 7777777777
+    steps:
+      - name: Do stuff
+      - name: Do stuff
+      - ...
+      - name: Slack Alert
+        uses: smartcontractkit/.github/actions/pr-slack-alert@{sha-tag} # 0.1.0
```
Comment (erikburt) [inline actions/pr-slack-alert/README.md:24]:
```suggestion
        uses: smartcontractkit/.github/actions/pr-slack-alert@{sha-tag} # pr-slack-alert@0.1.0
```

File: actions/pr-slack-alert/action.yml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout repo
+      if: inputs.checkout-repo == 'true'
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        fetch-depth: ${{ inputs.checkout-repo-fetch-depth }}
```
Comment (erikburt) [inline actions/pr-slack-alert/action.yml:69]: I don't know why this is in the boiler-plate. Is there a reason why we should include this here, and not let the action's users checkout the repo before calling this?

Treat this comment as non-blocking by the way, can be merged with this, just curious at your take on this.

File: actions/pr-slack-alert/action.yml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    description:
+      "will only send alert if the team name is included in the reviewers list"
+    required: false
+    default: ""
+  pr-event-type:
+    description:
+      "filter to trigger on specific PR event type - if set to empty it will
+      trigger on every type"
+    required: false
+    default: ""
```
Comment (erikburt) [inline actions/pr-slack-alert/action.yml:41]: Can we include some example types here and maybe some a github docs link for event types in the README?

File: actions/pr-slack-alert/payload.json [hunk 4]
```diff
@@ -0,0 +1,62 @@
+{
```
Comment (erikburt) [inline actions/pr-slack-alert/payload.json:1]: Seems like this payload will create a message for reviewing a PR, would it make sense to generify it based on the pr event type?
Doesn't need to be done now, could maybe add a section in the readme for future work.

**Review by erikburt — APPROVED** [review]
Good work 😄 left some small comments.

File: actions/pr-slack-alert/action.yml (hunk 2 above)
Comment (momentmaker) [reply actions/pr-slack-alert/action.yml:69]: yeah very true - I think we've just stuck with this pattern but never really questioned why lol

File: actions/pr-slack-alert/payload.json (hunk 4 above)
Comment (momentmaker) [reply actions/pr-slack-alert/payload.json:1]: ahh good call out!

File: actions/pr-slack-alert/action.yml (hunk 2 above)
Comment (momentmaker) [reply actions/pr-slack-alert/action.yml]: removed it :)

**Review by chainchad — APPROVED** [review]

**Review by erikburt — APPROVED** [review]

**Pull request merged**